### PR TITLE
Add audited noFailApiRequest method to BSDClient

### LINKED ISSUE #89: Handle failed BSD API calls
We need some sort of audit table for robert to go through.

### PR DESCRIPTION
**Pull request by schneidmaster**
Closes #89

### CODE REVIEW

File: src/backend/bsd.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import querystring from 'querystring';
 import {parseString} from 'xml2js';
 import Promise from 'bluebird';
 import qs from 'querystring';
+import GCAudit from './data/models/gc-audit';
```
Comment (saikat) [inline src/backend/bsd.js]: Would actually make it BSDAudit.  The convention I'm following is BSDBlah is stuff related to BSD, GCBSDBlah are extended models (so GCBSDGroup has an underlying BSDGroup).  We may one day have a VANAudit as well.

File: src/backend/bsd.js [hunk 2]
```diff
@@ -17,6 +18,19 @@ export default class BSD {
...
   }
 
+  noFailApiRequest(method, ...args) {
+    try {
+      BSD[method](...args);
+    } catch (e) {
+      GCAudit.create({
+        class: 'BSDClient',
+        method: method,
+        params: String(args),
```
Comment (saikat) [inline src/backend/bsd.js]: We may want to automatically retry these.  Is String good enough for serialization?

File: src/backend/data/models/gc-audit.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      type: DataTypes.STRING
+    },
+    params: {
+      type: DataTypes.STRING
+    },
+    error: {
+      type: DataTypes.STRING
+    }
+  }, {
+    underscored: true,
```
Comment (saikat) [inline src/backend/data/models/gc-audit.js]: I've just standardized on adding modified_dt and create_dt everywhere.  See the other models' createdAt and updatedAt methods.

File: src/backend/bsd.js (hunk 2 above)
Comment (schneidmaster) [inline src/backend/bsd.js]: Yup, `...args` is just an array. Do we want to auto retry just once?

File: src/backend/data/models/gc-audit.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+export default function(sequelize, DataTypes) {
+  let GCAudit = sequelize.define('GCAudit', {
+    id: {
```
Comment (saikat) [inline src/backend/data/models/gc-audit.js]: Not necessary, i think sequelize inserts this field.

File: src/backend/data/models/gc-audit.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      type: DataTypes.STRING
+    },
+    error: {
+      type: DataTypes.STRING
+    }
+  }, {
+    underscored: true,
+    tableName: 'gc_audits',
+    indexes: [
+      { fields: ['id'] }
```
Comment (saikat) [inline src/backend/data/models/gc-audit.js]: Also not necessary, primary keys are automatically indexed.

**Pull request merged**